src/app/components/import-annotations/details/details.component.ts

```
@@ -0,0 +1,333 @@
+  protected submitImportBuffer(): void {
+    // importing invalid annotation imports results in an internal server error
+    // we should therefore filter out any invalid models before importing
+    // however, this should be caught by form validation, preventing a user to import if there are errors
+    const validImportBuffers = this.importBuffer.filter(
+      (model) => model.errors.length === 0 && model.files?.length > 0
+    );
+
+    validImportBuffers.forEach((model: ImportGroup, i: number) =>
+      this.importEventGroup(model, i)
+    );
+  }
+
+  protected importEventGroup(model: ImportGroup, i: number): void {
+    // we have to do this because by default, a FileList is not an array, but uses an iterator
+    // therefore, if we convert it to an array, we can use a for-loop to import each file individually
+    const modelFiles: File[] = Array.from(model.files);
+
+    modelFiles.forEach((file: File) => {
+      const audioEventImportModel: AudioEventImportFileWrite =
+        new AudioEventImportFileWrite({
+          id: this.audioEventImport.id,
+          file,
+          additionalTagIds: model.additionalTagIds,
+          commit: true,
+        });
+
+      // because the event imports api implement the baw-api service, we don't need to catch errors here
+      this.eventImportsApi
+        .importFile(audioEventImportModel)
+        .pipe(takeUntil(this.unsubscribe))
+        .subscribe({
+          next: () => {
+            this.removeFromBuffer(i);
+            this.refreshTables();
+          },
+          error: (error: BawApiError) => {
+            // some of the default error messages are ambiguous on this page
+            // e.g. "you do not have access to this page" means that the user doesn't have access to the audio recording
+            if (error.status === FORBIDDEN) {
+              this.notifications.error(
+                "You do not have access to all the audio recordings or tags in your files."
+              );
```

we're effectively squashing an error here

- make sure the error message passed back from the server is included
- file a bug against the API
- cross link issue reference

---

Updated https://github.com/QutEcoacoustics/baw-server/issues/664 with the bug